File: 217_contains_duplicate/217_contains_duplicate.cpp
#include <unordered_map>
using namespace std;
class Solution
{
public:
	bool containsDuplicate(vector<int> &nums)
	{
		unordered_map<int, int> hashDuplicate;

		for (auto &&num : nums)
		{
			if (hashDuplicate.find(num) != hashDuplicate.end())
			{
				return true;
			}
			hashDuplicate[num] = 1;
		}
		return false;
	}
};

File: 217_contains_duplicate/217_contains_duplicate.java
import java.util.HashSet;

class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashSet<Integer> hashDuplicate = new HashSet<Integer>();
        for (Integer num : nums) {
            if (hashDuplicate.contains(num)) {
                return true;
            }
            hashDuplicate.add(num);
        }
        return false;

    }
}